shoreline.go

```diff
@@ -32,156 +33,75 @@ import (
 
 	"github.com/mdblp/shoreline/user"
 
-	common "github.com/tidepool-org/go-common"
-	"github.com/tidepool-org/go-common/clients"
-	"github.com/tidepool-org/go-common/clients/disc"
-	"github.com/tidepool-org/go-common/clients/hakken"
-	"github.com/tidepool-org/go-common/clients/mongo"
-)
-
-type (
-	// Config is the Shoreline main configuration
-	Config struct {
-		clients.Config
-		Service disc.ServiceListing `json:"service"`
-		Mongo   mongo.Config        `json:"mongo"`
-		User    user.ApiConfig      `json:"user"`
-	}
+	"github.com/mdblp/go-common/clients/mongo"
 )
 
 func main() {
-	var config Config
 	logger := log.New(os.Stdout, user.USER_API_PREFIX, log.LstdFlags|log.Lshortfile)
 	auditLogger := log.New(os.Stdout, user.USER_API_PREFIX, log.LstdFlags)
 	// Init random number generator
 	rand.Seed(time.Now().UnixNano())
 
-	// Set some default config values
-	config.User.MaxFailedLogin = 5
-	config.User.DelayBeforeNextLoginAttempt = 10 // 10 minutes
-	config.User.MaxConcurrentLogin = 100
-	config.User.BlockParallelLogin = true
-
-	if err := common.LoadEnvironmentConfig([]string{"TIDEPOOL_SHORELINE_ENV", "TIDEPOOL_SHORELINE_SERVICE"}, &config); err != nil {
-		logger.Panic("Problem loading Shoreline config", err)
-	}
-
-	// server secret may be passed via a separate env variable to accomodate easy secrets injection via Kubernetes
-	// The server secret is the password any Tidepool service is supposed to know and pass to shoreline for authentication and for getting token
-	// With Mdblp, we consider we can have different server secrets
-	// These secrets are hosted in a map[string][string] instead of single string
-	// which 1st string represents Server/Service name and 2nd represents the actual secret
-	// here we consider this SERVER_SECRET that can be injected via Kubernetes is the one for the default server/service (any Tidepool service)
-	serverSecret, found := os.LookupEnv("SERVER_SECRET")
-	if found {
-		config.User.ServerSecrets["default"] = serverSecret
-	}
-	// extract the list of token secrets
-	config.User.TokenSecrets = make(map[string]string)
-	zdkSecret, found := os.LookupEnv("ZENDESK_SECRET")
-	if found {
-		config.User.TokenSecrets["zendesk"] = zdkSecret
-	}
-	userSecret, found := os.LookupEnv("API_SECRET")
-	if found {
-		config.User.Secret = userSecret
-		config.User.TokenSecrets["default"] = userSecret
-	}
-
-	longTermKey, found := os.LookupEnv("LONG_TERM_KEY")
-	if found {
-		config.User.LongTermKey = longTermKey
-	}
+	var mongoConfig mongo.Config
 
-	verificationSecret, found := os.LookupEnv("VERIFICATION_SECRET")
-	if found {
-		config.User.VerificationSecret = verificationSecret
+	servicePort, found := os.LookupEnv("SHORELINE_PORT")
+	if !found || servicePort == "" {
```

Comment:
use os.Getenv that returns an empty string when not found
```suggestion
	servicePort := os.Getenv("SHORELINE_PORT")
	if servicePort == "" {
```

Reply:
True, suspenders belt test !
I can remove it if you prefer